Firmware/ultralcd.cpp

```diff
@@ -5297,81 +5299,87 @@ static void lcd_main_menu()
+                || lcd_commands_type != LcdCommands::Idle) {
+                if (!card.isFileOpen()) {
+                    if (!usb_timer.running() && (lcd_commands_type == LcdCommands::Idle)) {
+                        bMain=true;               // flag ('fake parameter') for 'lcd_sdcard_menu()' function
+                        MENU_ITEM_SUBMENU_P(_T(MSG_CARD_MENU), lcd_sdcard_menu);
+                    }
 #if SDCARDDETECT < 1
-            MENU_ITEM_GCODE_P(_i("Change SD card"), PSTR("M21"));  // SD-card changed by user ////MSG_CNG_SDCARD c=18
+             MENU_ITEM_GCODE_P(_i("Change SD card"), PSTR("M21"));  // SD-card changed by user ////MSG_CNG_SDCARD c=18
 #endif //SDCARDDETECT
-            }
-        } else {
-            bMain=true;                                   // flag (i.e. 'fake parameter') for 'lcd_sdcard_menu()' function
-            MENU_ITEM_SUBMENU_P(_i("No SD card"), lcd_sdcard_menu); ////MSG_NO_CARD c=18
+                }
+            } else {
+                bMain=true;                                   // flag (i.e. 'fake parameter') for 'lcd_sdcard_menu()' function
+                MENU_ITEM_BACK_P(_i("No SD card")); ////MSG_NO_CARD c=18
 #if SDCARDDETECT < 1
             MENU_ITEM_GCODE_P(_i("Init. SD card"), PSTR("M21")); // Manually initialize the SD-card via user interface ////MSG_INIT_SDCARD c=18
 #endif //SDCARDDETECT
-        }
+         }
 #endif //SDSUPPORT
-    }
-
-    if(!printer_active() && !farm_mode) {
-        const int8_t sheet = eeprom_read_byte(&(EEPROM_Sheets_base->active_sheet));
-        const int8_t nextSheet = eeprom_next_initialized_sheet(sheet);
-        if ((nextSheet >= 0) && (sheet != nextSheet)) { // show menu only if we have 2 or more sheets initialized
-            MENU_ITEM_FUNCTION_E(EEPROM_Sheets_base->s[sheet], eeprom_switch_to_next_sheet);
-        }
+        if(!printer_active() && !farm_mode) {
+            const int8_t sheet = eeprom_read_byte(&(EEPROM_Sheets_base->active_sheet));
+            const int8_t nextSheet = eeprom_next_initialized_sheet(sheet);
+            if ((nextSheet >= 0) && (sheet != nextSheet)) { // show menu only if we have 2 or more sheets initialized
+                MENU_ITEM_FUNCTION_E(EEPROM_Sheets_base->s[sheet], eeprom_switch_to_next_sheet);
+            }
 #ifdef QUICK_NOZZLE_CHANGE
-        SETTINGS_NOZZLE;
+            SETTINGS_NOZZLE;
 #endif //QUICK_NOZZLE_CHANGE
 
-    }
+        }
 
-    if ( ! ( printer_active() || (eFilamentAction != FilamentAction::None) || Stopped ) ) {
-        if (MMU2::mmu2.Enabled()) {
-            if(!MMU2::mmu2.FindaDetectsFilament() && !fsensor.getFilamentPresent()) {
-                // The MMU 'Load filament' state machine will reject the command if any 
-                // filament sensor is reporting a detected filament
-                MENU_ITEM_SUBMENU_P(_T(MSG_PRELOAD_TO_MMU), mmu_preload_filament_menu);
+        if (!( printer_active() || (eFilamentAction != FilamentAction::None) || Stopped )) {
```

Comment:
`!printer_active()` should always be true here since top-level if statement checks for it

Reply:
Right, thanks